### PR TITLE
Tuple missing types

### LINKED ISSUE #64: Missing 'name' and 'fullname' values in case of tuple-based TestCaseSource
Hello!

I have found the issue while trying to use your project in such specific cases:
```C#
namespace NUnitLoggerSampleProject {
    public class Tests {
        [TestCaseSource(nameof(GetInvalidTestCases))]
        public void InvalidCase((int, int) argument) {
            Assert.Pass();
        }

        static IEnumerable<(int, int)> GetInvalidTestCases() {
            return new[] {
                (0, 1)
            };
        }
    }
}
```
It leads to that result:
```Xml
<test-suite type="TestSuite" name="UnknownNamespace" fullname="UnknownNamespace" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" duration="0.031035">
    <test-suite type="TestFixture" name="UnknownType" fullname="UnknownNamespace.UnknownType" total="1" passed="1" failed="0" inconclusive="0" skipped="0" result="Passed" duration="0.031035">
        <test-case name="InvalidCase((0, 1))" fullname="UnknownNamespace.UnknownType.NUnitLoggerSampleProject.Tests.InvalidCase((0, 1))" methodname="NUnitLoggerSampleProject.Tests.InvalidCase((0, 1))" classname="UnknownType" result="Passed" duration="0.031035" asserts="0" />
    </test-suite>
</test-suite>
```
And the additional output in console:
```
nunitXml Logger: Unable to parse the test name 'NUnitLoggerSampleProject.Tests.InvalidCase((0, 1))' into a namespace type and method. Using Namespace='UnknownNamespace', Type='UnknownType' and Method='NUnitLoggerSampleProject.Tests.InvalidCase((0, 1))'
```
I made the reproduction project here - https://github.com/KonH/NUnitLoggerSampleProject

### PR DESCRIPTION
**Pull request by KonH**
Closes #64 
The proposed solution to track extra parentheses:
Instead of treat additional parentheses as an error in place, ensure that they are balanced (e.g., two closed and two open)
New tests, covers tuple case, included

### CODE REVIEW

File: test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs [hunk 1]
```diff
@@ -27,6 +27,10 @@ public class TestCaseNameParserTests
         // Tests with longer type and method names
         [DataRow("z.y.x.ape.bar(\"ar.g\",\"\\\"\")", "z.y.x", "ape", "bar(\"ar.g\",\"\\\"\")")]
         [DataRow("z.y.x.ape.bar(\"ar.g\",\")(\")", "z.y.x", "ape", "bar(\"ar.g\",\")(\")")]
+
+        // Test with tuple arguments
+        [DataRow("z.a.b((0,1))", "z", "a", "b((0,1))")]
+        [DataRow("z.a.b((\"arg\",1))", "z", "a", "b((\"arg\",1))")]
```
Comment (Siphonophora) [inline test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: @KonH Thanks for pointing this out. 

Sorry for the delay in reviewing this. Here are some more complex test cases which currently fail. If you want to work on fixing this let me know. Otherwise, I can take more of a look in a few days.

        [DataRow("z.a.b((0,1),(2,3))", "z", "a", "b((0,1),(2,3))")]
        [DataRow("z.a.b((0,(0,1)),(0,1))", "z", "a", "b((0,(0,1)),(0,1))")]

Comment (KonH) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: Thanks! I want, will be trying to fix it soon

Comment (Siphonophora) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: Sounds good, Thanks

Comment (KonH) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: Add more test cases and push solution for them, now tests passed
I'm not sure about other corner cases, but that approach looks suitable

Comment (KonH) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: But appveyor build failed for some reason (as you can see, dependencies aren't changed in that branch):
`C:\Program Files\dotnet\sdk\3.1.401\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(198,5): error NU5104: A stable release of a package should not have a prerelease dependency. Either modify the version spec of dependency "Stylecop.Analyzers [1.1.0-beta009, )" or update the version field in the nuspec. [C:\projects\nunit-testlogger\src\NUnit.Xml.TestLogger\NUnit.Xml.TestLogger.csproj]`

Comment (Siphonophora) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: @KonH Thanks. I'm happy with the changes. The appveyor issue is from our tooling. I think you won't need to make changes for that. 

@codito I think all 3 pulls are ready to be merged. #68 Adds the logo and would resolve all the appveyor issues. If we subsequently merge this and #67 then Master shouldn't have build issues.

Comment (Siphonophora) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: Actually, taking a look and #65 and #67 made some changes which aren't totally compatible. Let me sort that out

Comment (Siphonophora) [reply test/NUnit.Xml.TestLogger.UnitTests/TestCaseNameParserTests.cs:36]: Ok, so it was a simple fix when pulling it into the junit logger. https://github.com/spekt/junit.testlogger/pull/29/commits/94af6c504e522ea00b5f382706e5dd4352d48cf7

File: src/NUnit.Xml.TestLogger/TestCaseNameParser.cs [hunk 2]
```diff
@@ -123,7 +129,12 @@ public static ParsedName Parse(string fullyQualifedName)
                     }
                     else if (state == NameParseState.Parenthesis)
                     {
-                        if (thisChar == ')' || thisChar == '\\')
+                        if (thisChar == ')')
+                        {
+                            parenthesisCount++;
+                        }
+
+                        if (thisChar == '\\')
```
Comment (codito) [inline src/NUnit.Xml.TestLogger/TestCaseNameParser.cs]: With 70f258f, this condition may not be necessary. I believe build should be fixed too. Can we pull the latest master into this PR please?

Comment (KonH) [reply src/NUnit.Xml.TestLogger/TestCaseNameParser.cs]: I have rebased this PR branch on the actual master and remove that condition. But two tests now failed, so I add one more commit to cover failed cases in Parse_FailsGracefullyOnNonParsableInputs_WithConsoleOutput

Comment (KonH) [reply src/NUnit.Xml.TestLogger/TestCaseNameParser.cs]: And now CI builds finished successfully

**Review by codito — APPROVED** [review]
Thanks @KonH for the fix!
